go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -118,8 +118,8 @@ func (n *NodeController) addOrUpdatePod(pod *kapi.Pod, ignoreLearn bool) error {
 	if err != nil {
 		return err
 	}
-	namespaceExternalGw := namespace.GetAnnotations()[hotypes.HybridOverlayExternalGw]
-	namespaceVTEP := namespace.GetAnnotations()[hotypes.HybridOverlayVTEP]
+	namespaceExternalGwRaw := namespace.GetAnnotations()[hotypes.HybridOverlayExternalGw]
+	namespaceVTEPRaw := namespace.GetAnnotations()[hotypes.HybridOverlayVTEP]
```

Comment:
It's weird to read these here but not parse them until later. You should parse and bail out on error right away.

Also, it's weird to only log about the _Namespace_ error when updating a _Pod_. There should be checking in the `AddFunc` / `UpdateFunc` in `startNamespaceWatch`. If a namespace has a bad annotation, and we haven't logged about it yet, we should log an event (with an `EventRecorder`; there's an example in `pkg/ovn/ovn.go`).